### DISCUSSION TITLE
Spectator mode only client

### DISCUSSION THREAD

**Discussion by PabloDons** [iceiix/stevenarella#563] [Q&A]
Hello! I am starting a discussion here for a particular use case of stevenarella that I find extremely attractive! Just wondering what the scope would be and how much I'm really asking lol.

I am thinking about live streaming. Think thousands of viewers. Problem of course is that most minecraft servers can't handle anywhere near that many. But a service that only sends packets without processing any input is very lightweight and can potentially handle that much.

I imagine a client (for the web) that can only receive packets (not send). I also imagine a proprietary server mod that captures all packets sent to a player and forwards them to arbitrarily many clients. I have limited understanding of minecraft packets, but I imagine it possible to fly around in a client-side space generated by the packet data only, where the web client isn't actually logged in.

Is this possible with stevenarella? Can I somehow only provide it packets and it giving me a minecraft client to spectate the world around a player all in the browser?

This question is really independent of the state of stevenarella right now, but rather concerning the prospects. I imagine also such a client would require some coding to create, but is it a huge task or just flipping some switches and replacing some implementations?

**Comment by mpfaff** [comment]
Interesting idea! What you're describing is sort of like the freecam found in many utility mods and cheat clients in terms of implementation. Basically, the client would modify the local position and rotation without sending the respective packets to the server.

I think that with a server-side plugin, you could even connect without creating the entity on the server (not sure how it would tell between the normal client and the spectating one though. maybe by username or uuid, or by client brand?), which would eliminate what I believe to be the largest source of lag from a large number of players.

Edit: not sure if there's a way to "comment" here without "answering"
